src/Affjax.purs

```diff
@@ -168,8 +174,50 @@ patch_ url = map void <<< patch ResponseFormat.ignore url
 -- | get json "/resource"
 -- | ```
 request :: forall a. Request a -> Aff (Either Error (Response a))
-request req =
-  case req.content of
+request = \req ->
```

Comment:
Could you return this function to its prior formatting, ie.

```purs
request :: forall a. Request a -> Aff (Either Error (Response a))
request req = 
  case req.content of
    Nothing ->
      send (toNullable Nothing)
    Just content ->
      case extractContent content of
        Right c ->
          send (toNullable (Just c))
        Left err ->
          pure $ Left (RequestContentError err)
```

It's not necessary to move `ajaxRequest`, `send`, `headers`, and `fromResponse` out of the `where` clause where they were before (I tested moving them back locally and things compile just fine). Can you move them back? That will minimize the diff and prevent us from splitting some functions into `let` bindings and others underneath `where`.

Reply:
Ok, I write why I did this - for speed and makes easier to see what is dependent on args and what is not

Fix anyway?

Reply:
@thomashoneyman

Reply:
I haven't examined the output code -- in what way does this improve performance? I'm genuinely curious.

Reply:
currently it compiles to

https://pastebin.com/PFU8femC

now

```
var request = (function () {
    var notDependentOnArg1 = function (v) { ... }
    var notDependentOnArg2 = function (v) { ... }
    return function(req) {
      var dependentOnArg1 = function (v) { ... req ... }
      var dependentOnArg2 = function (v) { ... req ... }

      ...
    }
})()
```

before

```
var request = function(req) {
  var notDependentOnArg1 = function (v) { ... }
  var notDependentOnArg2 = function (v) { ... }

  var dependentOnArg1 = function (v) { ... req ... }
  var dependentOnArg2 = function (v) { ... req ... }

  ...
}
```

@thomashoneyman

Reply:
> Premature optimization is the root of all evil (or at least most of it) in programming

This is code that is always run in conjunction with a _network request_. You'll never be able to notice any difference here because network latency dominates a couple of closure allocations, so readability/uniformity has way higher priority.

I can appreciate the argument about making the scoping clearer, but I don't think mixing `let` and `where` like that actually does that, because you have to read from the top _and_ bottom to find the sandwiched piece of logic in the middle.